WcaOnRails/app/models/competition.rb

```diff
@@ -222,6 +235,7 @@ def update_foreign_keys_when_id_changes
       CompetitionMedium.where(competitionId: id_was).update_all(competitionId: id)
       CompetitionDelegate.where(competition_id: id_was).update_all(competition_id: id)
       CompetitionOrganizer.where(competition_id: id_was).update_all(competition_id: id)
+      DelegateReport.where(competition_id: id_was).update_all(competition_id: id)
```

Comment:
This is a bit messy, would you like the following?
```ruby
  after_save :update_foreign_keys_when_id_changes
  def update_foreign_keys_when_id_changes
    if id_changed?
      [Result, Registration, Scramble, CompetitionMedium, CompetitionDelegate, CompetitionOrganizer, DelegateReport].each do |model|
        foreign_key = model.column_names.include?("competitionId") ? "competitionId" : "competition_id"
        model.where(foreign_key => id_was).update_all(foreign_key => id)
      end
    end
  end
```

Also, personally I think that `after_save :update_foreign_keys, if: :id_changed?` would nicely replace the current name but as I remember you prefer not to use this DSL, right?